index.bs

```diff
@@ -1289,52 +1308,105 @@ SessionResult = (StatusResult)
 
 <div algorithm>
 
-To <dfn lt="updating the event map">update the event map</dfn>, given
-|session|, |list of event names|, |list of contexts|, and |enabled|:
+To <dfn>update the event map</dfn>, given
+|session|, |list of event names|, |browsing contexts|, and |enabled|:
 
-  1. Let |global event set| be the [=global event set=] for |session|.
+Note: The return value of this algorithm is a map between event names and
+contexts. When the events are being enabled globally, the contexts in the return
+value are those for which the event was already enabled. When the events are
+enabled for specific contexts, the contexts in the return value are those for
+which the event are now enabled but were not previously. When events are
+disabled, the return value is always empty.
 
-  1. Let |event map| be the [=browsing context event map=] for |session|.
+  1. Let |global event set| be a [=set/clone=] of the [=global event set=] for
+     |session|.
+
+  1. Let |event map| be a [=map/clone=] the [=browsing context event map=] for
+     |session|.
+
+  1. Let |enabled events| be a new map.
 
   1. Let |event names| be an empty set.
 
-    1. For each entry |name| in the |list of event names|, let |event names| be
+    1. For each entry |name| in |list of event names|, let |event names| be
        the union of |event names| and the result of [=trying=] to [=obtain a set
        of event names=] with |name|.
 
-    1. If the |list of contexts| is null:
+  1. If |browsing contexts| is null:
+
+      1. If |enabled| is true:
 
-      1. If |enabled| is true, for each |event name| in |event names|,
-         append |event name| to |global event set|. Otherwise for for each
-         |event name| in |event names|, if the |global event set| [=contains=]
-         |event name|, remove |event name| from the |global event set|.
+        1. For each |event name| of |event names|:
 
-      1. Return
+           1. If |global event set| doesn't contain |event name|:
 
-    1. Let |targets| be an empty list.
+             1. Let |event enabled contexts| be the [=event enabled browsing
+                contexts=] given |session| and |event name|
 
-    1. For each entry |context id| in the |list of contexts|:
+             1. |event name| to |global event set|.
+
+             1. For each |context| of |event enabled contexts|, remove |event
+                name| from |event map|[|context|].
+
+             1. Set |enabled events|[|event name|] to |event enabled contexts|.
+
+      1. If |enabled| is false:
+
+        1. For each |event name| in |event names|:
+
+          1. If the |global event set| [=contains=] |event name|, remove |event
```

Comment:
Nit: This step should refer to the local variable named "global event set", and not **the** global event set.